src/src/misc/error.js

```diff
@@ -0,0 +1,44 @@
+export class ErrorOptions {
+  msg = "";
+  handler = (params) => {};
+}
+
+class ErrorTriggerRequest {
+  httpResponse = null;
+  error = null;
+  msg = null;
+  handler = null;
+  showError = null;
+}
+
+class ErrorTriggerRequestBuilder {
+  val = new ErrorTriggerRequest();
+
+  setHttpResponse(input) {
+    this.val.httpResponse = input;
+    return this;
+  }
+
+  setError(input) {
+    this.val.error = input;
+    return this;
+  }
+
+  setMsg(input) {
+    this.val.msg = input;
+    return this;
+  }
+
+  setHandler(input) {
+    this.val.handler = input;
+    return this;
+  }
+
+  build() {
+    return this.val;
+  }
+}
```

Comment:
Not sure if using null like this is idiomatic javascript, but it surely will make code harder to write because you have to have null checks everywhere. Is there a way to rewrite this so that we can get rid of null values and checks?

Reply:
Agree that it's not ideal, but can't currently think of a nicer alternative. 

If all the values that is currently assigned to null is given a default value, I'd instead have to check all of them in a different way. If httpResponse has a default value, I need to make sure that the object contains the keys I'd expect from a properly filled out httpResponse before using it. If handler is given a default value, I can't know for sure whether the handler has been replaced by the user, thus giving it higher priority in the error handling flow. Well, I could build a different way of determining whether it's been switched out or not, but null also works and is much less effort.

Given the nature of how ErrorContext is used, there will almost always be some values that shouldn't be set. If a network error happens, then _httpResponse_ should never be populated, instead _error_ would be. In most cases, if _httpResponse_ is populated, then _error_ rarely is. _msg_  and _handler_ can be populated regardless of any of these, that would be up to the user to set that one in most cases. _showError_ is probably the only one that you can almost always rely on being available, since it is set by the useError Hook.

Reply:
Wow a whole novel :D I was just curious, don't read too much into my comment.